src/main/java/seedu/address/model/person/RoomNumberPredicate.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+
+
+/**
+ * A predicate class that tests whether a Person's room number matches a given room number.
+ */
+public class RoomNumberPredicate implements Predicate<Person> {
+    private final String roomNumber;
+
+
+
+    /**
+     * Constructs a new RoomNumberPredicate with a specified room number.
+     *
+     * @param roomNumber The room number to compare against Person objects.
+     */
+    public RoomNumberPredicate(String roomNumber) {
+        this.roomNumber = roomNumber;
+    }
+
+
+    /**
+     * Evaluates this predicate on the given Person instance.
+     * Tests if the Person's room number matches the room number provided during instantiation.
+     *
+     * @param person The Person to be tested against this predicate.
+     * @return true if the Person's room number matches the predicate's room number, otherwise false.
+     */
+    @Override
+    public boolean test(Person person) {
+        return person.getRoomNumber().toString().equals(roomNumber);
```

Comment:
Just adding on to yf's comment, would it be better if this is saved as a RoomNumber object and then we can use person.getRoomNumber().equls(roomNumber) in the test function?